src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -50,27 +119,343 @@ def execute(self, reqStatus):
+            msg += "\nNumber of properly cleaned workflows: %s."
+            msg += "\nNumber of archived workflows: %s."
+            self.logger.info(msg,
+                             totalNumRequests,
+                             cleanNumRequests,
+                             archivedNumRequests)
+            self.updateReportDict(summary, "total_num_requests", totalNumRequests)
+            self.updateReportDict(summary, "clean_num_requests", cleanNumRequests)
+            self.updateReportDict(summary, "archived_num_requests", archivedNumRequests)
+        except Exception as ex:
+            msg = "Unknown exception while running MSRuleCleaner thread Error: %s"
+            self.logger.exception(msg, str(ex))
+            self.updateReportDict(summary, "error", msg)
+
         return summary
 
+    def _execute(self, reqRecords):
+        """
+        Executes the MSRuleCleaner pipelines based on the workflow status
+        :param reqList: A list of RequestRecords to work on
+        :return:        a tuple with:
+                            number of properly cleaned requests
+                            number of processed workflows
+                            number of archived workflows
+        """
+        # NOTE: The Input Cleanup, the Block Level Cleanup and the Archival
+        #       Pipelines are executed sequentially in the above order.
+        #       This way we assure ourselves that we archive only workflows
+        #       that have accomplished the needed cleanup
+
+        cleanNumRequests = 0
+        totalNumRequests = 0
+
+        # Call the workflow dispatcher:
+        for _, req in reqRecords.items():
+            wflow = MSRuleCleanerWflow(req)
+            self._dispatchWflow(wflow)
+            msg = "\n----------------------------------------------------------"
+            msg += "\nMSRuleCleanerWflow: %s"
+            msg += "\n----------------------------------------------------------"
+            self.logger.debug(msg, pformat(wflow))
+            totalNumRequests += 1
+            if self._checkClean(wflow):
+                cleanNumRequests += 1
+
+        # Report the counters:
+        for pline in self.cleanuplines:
+            msg = "Workflows cleaned by pipeline: %s: %d"
+            self.logger.info(msg, pline.name, self.wfCounters['cleaned'][pline.name])
+        archivedNumRequests = self.wfCounters['archived']
+        self.logger.info("Workflows archived: %d", self.wfCounters['archived'])
+        return totalNumRequests, cleanNumRequests, archivedNumRequests
+
+    def _dispatchWflow(self, wflow):
+        """
+        A function intended to dispatch a workflow (e.g based on its status)
+        through one or more functional pipelines in case there is some more
+        complicated logic involved in the order we execute them but not just
+        a sequentially
+        """
+        # msg = "Called with wflow: %s"
+        # self.logger.debug(msg, pformat(wflow))
```

Comment:
How about using this log, just to say which workflow is getting into the pipeline, something like:
```
self.loger.info("Dispatching workflow: %s", wflow['RequestName'])
```

or `Working on workflow: ....`

Reply:
that's a good idea ... I was using already using it in a similar way for some investigation, but forgot to remove it
Done now